storage.go

```diff
@@ -3,34 +3,170 @@ package ftp
 import (
 	"context"
 	"io"
+	"net/textproto"
+	"path/filepath"
 
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
+	"github.com/beyondstorage/go-storage/v4/services"
 	. "github.com/beyondstorage/go-storage/v4/types"
+	"github.com/jlaffaye/ftp"
+	mime "github.com/qingstor/go-mime"
 )
 
+type listDirInput struct {
+	rp  string
+	dir string
+
+	started           bool
+	continuationToken string
+	objList           []*ftp.Entry
+	counter           int
+}
+
+func (input *listDirInput) ContinuationToken() string {
+	return input.continuationToken
+}
+
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(false)
+		o.Mode = ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode = ModeRead
+	}
+
+	o.ID = filepath.Join(s.workDir, path)
+	o.Path = path
+	return o
+}
+
+func (s *Storage) createDir(ctx context.Context, path string) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+	err = s.connection.MakeDir(rp)
+	if err != nil {
+		return nil, err
+	}
+
+	o = s.newObject(true)
+	o.ID = rp
+	o.Path = path
+	o.Mode |= ModeDir
+	return
 }
 
 func (s *Storage) delete(ctx context.Context, path string, opt pairStorageDelete) (err error) {
-	panic("not implemented")
+	rp := s.getAbsPath(path)
+	err = s.connection.Delete(rp)
+	if err != nil && err.(*textproto.Error).Code != ftp.StatusFileUnavailable {
+		return err
+	}
```

Comment:
We should check ok when type assert, otherwise this would panic if type assert failed. Moreover, I'd like to call `errors.As` rather than type assert directly.

So here is my suggestion:
```suggestion
	if err != nil {
		var txtErr *textproto.Error
		// ignore error with code ftp.StatusFileUnavailable, to make delete idempotent
		if errors.As(err, &txtErr) && txtErr.Code == ftp.StatusFileUnavailable {
			return nil
		}
		return err
	}
```